### PR TITLE
MF-2 - Use User Auth keys

### LINKED ISSUE #2: Use User Auth keys
Use User Auth keys instead of user credentials.
Resolved by #4

### PR DESCRIPTION
**Pull request by mteodor**
Resolves #2 
Signed-off-by: Mirko Teodorovic <mirko.teodorovic@gmail.com>

### CODE REVIEW

File: service.go [hunk 1]
```diff
@@ -76,10 +80,14 @@ func (ps *provisionService) Provision(externalID, externalKey string) (res Resul
 	var newThingID, ctrlChanID, dataChanID, token string
 	defer ps.recover(&err, &newThingID, &ctrlChanID, &dataChanID, &token)
 	channels := make([]string, 0)
+	things := make([]string, 0)
 
-	token, err = ps.sdk.CreateToken(ps.mfEmail, ps.mfPass)
-	if err != nil {
-		return res, err
+	token = ps.mfApiKey
+	if ps.mfApiKey == "" {
```
Comment (drasko) [inline service.go:86]: Why not `if token == ""`?

File: service.go [hunk 2]
```diff
@@ -90,10 +98,12 @@ func (ps *provisionService) Provision(externalID, externalKey string) (res Resul
 	// Get newly created thing (in order to get the key).
 	thingCreated, err := ps.sdk.Thing(newThingID, token)
 	if err != nil {
-		return res, provsdk.ErrGetThing
+		return res, errors.Wrap(provsdk.ErrGetThing, fmt.Sprintf("thing id:%s", thingCreated.ID))
```
Comment (manuio) [inline service.go:101]: This log will be hard to read without a white space here `id:%s`

File: service.go [hunk 3]
```diff
@@ -106,12 +116,15 @@ func (ps *provisionService) Provision(externalID, externalKey string) (res Resul
...
-		if err != nil {
-			return res, provsdk.ErrConn
+	things = append(things, ps.predefinedThings...)
+
+	for _, c := range channels {
+		for _, t := range things {
+			// Connect predefined Things to control channel.
+			err = ps.sdk.Connect(t, c, token)
+			if err != nil {
+				return res, errors.Wrap(provsdk.ErrConn, fmt.Sprintf("ch:%s,th:%s", c, t))
```
Comment (manuio) [inline service.go:126]: same here

File: cmd/main.go [hunk 4]
```diff
@@ -130,6 +132,11 @@ func loadConfig() config {
 	if autoWhiteList && !provisionBS {
 		log.Fatalf("Can't auto whitelist if auto config save is off")
 	}
+	predefinedThings := make([]string, 0)
+	thingIDs := mainflux.Env(envThingIDs, defThingIDs)
```
Comment (drasko) [inline cmd/main.go:136]: This one is used only in `if`, so embed it's declaration in the line below.

File: service.go [hunk 5]
```diff
@@ -90,10 +98,12 @@ func (ps *provisionService) Provision(externalID, externalKey string) (res Resul
 	// Get newly created thing (in order to get the key).
 	thingCreated, err := ps.sdk.Thing(newThingID, token)
 	if err != nil {
-		return res, provsdk.ErrGetThing
+		return res, errors.Wrap(provsdk.ErrGetThing, fmt.Sprintf("thing id: %s", thingCreated.ID))
 	}
+	things = append(things, thingCreated.ID)
 
 	ctrlChannel, err := ps.sdk.CreateChannel("ctrlchan", "control", token)
+
```
Comment (drasko) [inline service.go:106]: Kill this white line

File: cmd/main.go [hunk 6]
```diff
@@ -130,6 +132,10 @@ func loadConfig() config {
 	if autoWhiteList && !provisionBS {
 		log.Fatalf("Can't auto whitelist if auto config save is off")
 	}
+	predefinedThings := make([]string, 0)
+	if len(mainflux.Env(envThingIDs, defThingIDs)) > 0 {
```
Comment (drasko) [inline cmd/main.go:136]: This `mainflux.Env(envThingIDs, defThingIDs)` is used two times. PLease revert to previous implementation, but embed declaration of `thingIDs` into `if`

**Pull request merged**